rust/blockstore/src/arrow/block/types.rs

```diff
@@ -9,15 +11,46 @@ use arrow::{
     array::{Array, StringArray},
     record_batch::RecordBatch,
 };
-use chroma_cache::cache::Cacheable;
 use chroma_error::{ChromaError, ErrorCodes};
-use std::cmp::Ordering::{Equal, Greater, Less};
-use std::io::SeekFrom;
 use thiserror::Error;
 use uuid::Uuid;
 
+use super::delta::BlockDelta;
+
 const ARROW_ALIGNMENT: usize = 64;
 
+/// A cached block is what sits in the cache.  We can create a Block from a cached block.
+#[derive(Clone, Debug, Default, serde::Serialize, serde::Deserialize)]
+pub struct CachedBlock {
+    pub id: Uuid,
+    pub data: Vec<u8>,
+}
+
+impl TryFrom<&Block> for CachedBlock {
+    type Error = crate::arrow::block::types::BlockToBytesError;
+
+    fn try_from(b: &Block) -> Result<Self, Self::Error> {
+        Ok(CachedBlock {
+            id: b.id,
+            data: b.to_bytes()?,
```

Comment:
I think from_bytes/to_bytes will result in copies of the data, do we care?

Reply:
We do.  I'm debating on a two-level cache for persistent and non-persistent cache to avoid this problem, but I don't know how we'll do that.